lib/fs.js

```diff
@@ -385,6 +386,12 @@ fs.readFile = function(path, options, callback) {
   if (!nullCheck(path, callback))
     return;
 
+  // doing justice between linux and windows
+  // for `/` char at the end of file path
+  if (isWindows && (path.slice(-1) === '/')) {
+    throw new errors.Error('ERR_INVALID_FILE_URL_PATH', path);
```

Comment:
1. Using `errors.TypeError` might be more appropriate.
2. The second parameter (here is `path`) of `ERR_INVALID_FILE_URL_PATH ` is wrong. It should be a string contains an error description about the invalid path.
Here is some example:
https://github.com/nodejs/node/blob/abe14ab24a56a94c6b37a4bda69e1179c39ce4ab/lib/internal/url.js#L1362-L1363
https://github.com/nodejs/node/blob/abe14ab24a56a94c6b37a4bda69e1179c39ce4ab/lib/internal/url.js#L1346-L1347

Reply:
I don't think `errors.TypeError` is appropriate here. A `TypeError` would be if we were expecting a string but received something else. In this case, we are expecting a string and receiving a string, but it has a value that is not valid. (By all means, please correct me if I'm wrong or misunderstanding something!)

The second point (about the second parameter) is :+1:.

Reply:
Now the similar `ERR_INVALID_FILE_URL_PATH` errors in `internal/url.js` are using  `TypeError` (some examples showed above).

I do agree with your point. For semantics, using `Error` is better.

Reply:
Thanks for the correction it's effected. I am still not clear about the error type @Trott .
From what @starkwang showed above, Please I might also be wrong but I think `TypeError` is appropriate for the case. 
errors in `internal/url.js` are expecting characters that should not include some special characters, The type there might be referring to character type not data type.